cli/src/pcluster/config/cluster_config.py

```diff
@@ -159,66 +205,79 @@ def __init__(self, queues_config: List[QueueConfig], scheduler: str = "slurm"):
         self.queues_config = queues_config
 
 
-class SharedStorageType(Enum):
-    """Describe the Type of a shared storage."""
-
-    EBS = "EBS"
-    EFS = "EFS"
-    FSX = "FSX"
-
-    @classmethod
-    def is_valid(cls, value):
-        """Verify if the given value is a valid SharedStorageType."""
-        return value in cls._member_names_
-
+class RaidConfig(Config):
+    """Represent the Raid configuration."""
 
-class SharedStorageConfig(Config):
-    """Represent a generic shared storage configuration."""
-
-    def __init__(self, mount_dir: str, storage_type: SharedStorageType):
+    def __init__(self, type: str = None, number_of_volumes=MyInt(2)):
         super().__init__()
-        self.type = storage_type
-        self.mount_dir = mount_dir
+        self.type = type
+        self.number_of_volumes = number_of_volumes
 
 
-class EbsConfig(SharedStorageConfig):
+class EbsConfig(Config):
     """Represent the EBS configuration."""
 
     def __init__(
         self,
-        mount_dir: str,
-        volume_type: str = None,
+        volume_type: str = MyStr("gp2"),
         iops: int = None,
-        size: int = None,
+        size: int = MyInt(20),
         encrypted: bool = None,
         kms_key_id: str = None,
         snapshot_id: str = None,
         id: str = None,
+        throughput: int = None,
+        raid: RaidConfig = None,
     ):
-        super().__init__(mount_dir, SharedStorageType.EBS)
+        super().__init__()
         self.volume_type = volume_type
+        if iops is None:
+            if volume_type in EBS_VOLUME_TYPE_IOPS_DEFAULT:
+                iops = EBS_VOLUME_TYPE_IOPS_DEFAULT.get(volume_type)
         self.iops = iops
         self.size = size
         self.encrypted = encrypted
         self.kms_key_id = kms_key_id
         self.snapshot_id = snapshot_id
         self.id = id
+        if throughput is None:
```

Comment:
minor: `if throughput is None and volume_type == "gp3":`

Reply:
Done